### PR TITLE
[Cosmos] Updates to kwargs and docstrings

### PR DESCRIPTION
**Pull request by annatisch**
Partial fix for #7729 and #7788 
Also address #8278

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/cosmos/azure-cosmos/azure/cosmos/_base.py [hunk 1]
```diff
@@ -63,20 +65,50 @@
...
-    'populate_quota_info': 'populateQuotaInfo'
+    'populate_quota_info': 'populateQuotaInfo',
+    'content_type': 'contentType',
+    'is_query_plan_request': 'isQueryPlanRequest',
+    'supported_query_features': 'supportedQueryFeatures',
+    'query_version': 'queryVersion'
 }
 
+def _get_match_headers(kwargs, match_param, etag_param):
+    # type: (str) -> Tuple(Dict[str, Any], Optional[str], Optional[str])
```
Comment (bryevdv) [inline sdk/cosmos/azure-cosmos/azure/cosmos/_base.py]: Optional?

File: sdk/cosmos/azure-cosmos/test/crud_tests.py [hunk 2]
```diff
@@ -859,7 +860,8 @@ def test_document_crud(self):
 
         # should pass for most recent etag
         replaced_document_conditional = created_collection.replace_item(
```
Comment (srinathnarayanan) [inline sdk/cosmos/azure-cosmos/test/crud_tests.py]: We should probably add some tests to cover the new errors being thrown when match_condition is specified without etag and other cases.

Comment (annatisch) [reply sdk/cosmos/azure-cosmos/test/crud_tests.py]: Added :)

**Pull request merged**